api/tests/unit_tests/core/tools/test_tool_parameter_converter.py

```diff
@@ -0,0 +1,56 @@
+import pytest
+
+from core.tools.entities.tool_entities import ToolParameter
+from core.tools.utils.tool_parameter_converter import ToolParameterConverter
+
+
+def test_get_parameter_type():
+    assert ToolParameterConverter.get_parameter_type(ToolParameter.ToolParameterType.STRING) == 'string'
+    assert ToolParameterConverter.get_parameter_type(ToolParameter.ToolParameterType.SELECT) == 'string'
+    assert ToolParameterConverter.get_parameter_type(ToolParameter.ToolParameterType.BOOLEAN) == 'boolean'
+    assert ToolParameterConverter.get_parameter_type(ToolParameter.ToolParameterType.NUMBER) == 'number'
+    with pytest.raises(ValueError):
+        ToolParameterConverter.get_parameter_type('unsupported_type')
+
+
+def test_cast_parameter_by_type():
+    # string
+    assert ToolParameterConverter.cast_parameter_by_type('test', ToolParameter.ToolParameterType.STRING) == 'test'
+    assert ToolParameterConverter.cast_parameter_by_type(1, ToolParameter.ToolParameterType.STRING) == '1'
+    assert ToolParameterConverter.cast_parameter_by_type(1.0, ToolParameter.ToolParameterType.STRING) == '1.0'
+    assert ToolParameterConverter.cast_parameter_by_type(None, ToolParameter.ToolParameterType.STRING) == ''
+
+    # secret input
+    assert ToolParameterConverter.cast_parameter_by_type('test', ToolParameter.ToolParameterType.SECRET_INPUT) == 'test'
+    assert ToolParameterConverter.cast_parameter_by_type(1, ToolParameter.ToolParameterType.SECRET_INPUT) == '1'
+    assert ToolParameterConverter.cast_parameter_by_type(1.0, ToolParameter.ToolParameterType.SECRET_INPUT) == '1.0'
+    assert ToolParameterConverter.cast_parameter_by_type(None, ToolParameter.ToolParameterType.SECRET_INPUT) == ''
+
+    # select
+    assert ToolParameterConverter.cast_parameter_by_type('test', ToolParameter.ToolParameterType.SELECT) == 'test'
+    assert ToolParameterConverter.cast_parameter_by_type(1, ToolParameter.ToolParameterType.SELECT) == '1'
+    assert ToolParameterConverter.cast_parameter_by_type(1.0, ToolParameter.ToolParameterType.SELECT) == '1.0'
+    assert ToolParameterConverter.cast_parameter_by_type(None, ToolParameter.ToolParameterType.SELECT) == ''
+
+    # boolean
+    true_values = [True, 'True', 'true', '1', 'YES', 'Yes', 'yes', 'y', 'something']
```

Comment:
Hi, any suggestion for handling unexpected boolean values like 'something'?
Currently, it's treated and converted to True. @Yeuoly

Reply:
Have already talked about this offline